components/AnnotationChart.tsx

```diff
@@ -0,0 +1,415 @@
+import React, {useState, useEffect, useRef} from 'react';
+import { staticVerticalLineColor, dynamicVerticalLineColor, staticVerticalLine, dynamicVerticalLine } from '../utils/utils';
+import { useStateSetters, useStates } from './AppContext'; 
+import { Bar } from 'react-chartjs-2';
+import {
+    Chart as ChartJS,
+    ChartData,
+    ChartDataset,
+    ChartOptions,
+    CategoryScale,
+    LinearScale,
+    PointElement,
+    LineElement,
+    BarElement,
+    Tooltip,
+    Legend,
+    Scale,
+    ChartType,
+  } from 'chart.js';
+  
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    PointElement,
+    LineElement,
+    BarElement,
+    Tooltip,
+    Legend,
+    staticVerticalLine,
+    dynamicVerticalLine,
+);
+
+
+
+/**
+ *  The current design for the chart is to only display one group of mutuallyExclusive category labels, such as 'chase' and 'no-chase'.
+ *  If need to display multiple groups of labels, such as 'chase'/'no-chase' and 'grooming'/'no-grooming', then need to use another instance of this comp.
+ *  Assume these labels are defined in one btnGroup, and each label is globally unique among all category btns.
+ *  Otherwise, it won't function correctly.
+ *  This comp should only be used for video annotation 
+ * 
+ *  props:
+ *      labels: ['chase', 'no-chase'], an arr of mutuallyExclusive labels to display on chart. Consistent with labels of annotation buttons. TODO: should be replaced by another ChartController component
+        width: str. '100%'/'200px'/'50vw'..., default is 100%. Set the width of the chart.
+        height: str. '100%'/'200px'/'50vh'..., default is 100%. Set the height of the chart
+        // legendPosition: 'top'/'left'/'bottom'/'right'/'chartArea'. 'bottom' by default. Position of legend.
+        // legendAlign: 'start'/'center'/'end'. 'end' by default. Horizontal position of legend.
+        staticVerticalLineColor: 'rgb()', '#xxxxxx', 'red'. The color of the static vertical Line
+        dynamicVerticalLineColor:  'rgb()', '#xxxxxx', 'red'. The color of the dynamic vertical Line
+        // zoomSpeed: number. Speed for zooming on y axis. 0.01 by default.
+        omitXLabels: boolean. Omit x-axis labels. false by default.
+        */
+
+interface AnnotationData {
+    frameNum: number | null,
+    range: number[] | null,
+    data: any
+}
+
+interface AnnotationChartProps {
+    labels: string[],
+    width?: string,
+    height?: string,
+    staticVerticalLineColor?: string, //'rgb()', '#xxxxxx', 'red'
+    dynamicVerticalLineColor?: string, //'rgb()', '#xxxxxx', 'red'
+    omitXLabels?: boolean,
+    legendPosition?: "bottom" | "left" | "top" | "right" | "center" | "chartArea" | undefined, // should include _DeepPartialObject<{ [scaleId: string]: number; }> ?
+    legendAlign?: "center" | "end" | "start" | undefined,
+}
+
+
+
+declare module 'chart.js' {
+   interface PluginOptionsByType<TType extends ChartType = ChartType> {
```

Comment:
Here you have staticVerticalLine and dynamicVerticalLine as two different types. It looks like dynamicVerticalLine could be a staticVerticalLine with an extra clickHandler property, if position and startIndex were rationalized to have the same property name. Is that possible?